chat-plugins/uno.js

```diff
@@ -0,0 +1,694 @@
+	message = message ? "<font color=\"red\"><b>" + message + "</b></font>" : "";
+	return html + "<table border=1 style=\"border-collapse: collapse;\"><tr><td>&nbsp;<b>Top Card</b>&nbsp;</td><td>&nbsp;<b>Your Hand</b>&nbsp;</td></tr>" + "<tr><td>" + topCard + "</td><td>" + yourHand + "</td></tr><tr><td>" + (pass ? passButton : drawButton) + "</td><td>" + message + "</td></tr></table>";
+}
+
+//for wildcards that summon a colour change
+function getColourChange(buffer, hand, uhtmlid) {
+	return "|uhtmlchange|" + uhtmlid + "|<table border=1 style=\"border-collapse: collapse;\"><tr><td>Choose which colour to change to:<br>" + '<button style="background: red; border: 2px solid rgba(33 , 68 , 72 , 0.59) ; width: 80px ; border-radius: 5px , auto" name="send" value="' + buffer + ' R"><font color="white" size=4>Red</font></button></center>' + '<button style="background: yellow; border: 2px solid rgba(33 , 68 , 72 , 0.59) ; width: 80px ; border-radius: 5px , auto" name="send" value="' + buffer + ' Y"><font color="black" size=4>Yellow</font></button></center>' + '<button style="background: blue; border: 2px solid rgba(33 , 68 , 72 , 0.59) ; width: 80px ; border-radius: 5px , auto" name="send" value="' + buffer + ' B"><font color="white" size=4>Blue</font></button></center>' + '<button style="background: green; border: 2px solid rgba(33 , 68 , 72 , 0.59) ; width: 80px ; border-radius: 5px , auto" name="send" value="' + buffer + ' G"><font color="white" size=4>Green</font></button></td></tr>' + '<tr><td>Your Cards: <br>' + buildHand(hand, true) + '</td></tr></table>';
+}
+
+function getUserName(id) {
+	return Users(id) ? Users(id).name : id;
+}
+
+class Game {
+	constructor(room, playerCap) {
+		if (!room.unoGameId) {
+			room.unoGameId = 0;
+		}
+		room.unoGameId++;
+		this.room = room;
+		this.id = 0;
+		this.top = null;
+		this.discard = [];
+		this.deck = null;
+		//for player order
+		this.list = [];
+		this.data = {};
+		this.joinedIps = {};
+		this.player = null;
+		this.gameid = "uno";
+		this.started = false;
+		this.timer = null;
+		this.lastDraw = null;
+		this.colourChange = null;
+		this.uhtmlChange = null;
+		this.playerCap = playerCap ? playerCap : null;
+	}
+
+	init() {
+		this.room.add("|uhtml|new" + this.room.unoGameId + "|<div class=\"broadcast-green\"><center><img src=\"http://www.theboardgamefamily.com/wp-content/uploads/2010/12/uno-mobile-game1.jpg\" height=150 width=160><br><font color=white><b>A new game of UNO is starting</b></font><br><button style=\"height: 25px ; width: 50px ;\" name=\"send\" value=\"/uno join\">Join</button></center></div>");
+	}
+
+	join(user) {
+		if (this.started) return false;
+		let ip = user.latestIp;
+		let userid = user.userid;
+
+		if (this.playerCap && this.list.length >= this.playerCap) {
+			return user.sendTo(this.room, "The game is already full.");
+		}
+
+		if (userid in this.data || ip in this.joinedIps) {
+			return user.sendTo(this.room, "You have already have an alt joined.");
```

Comment:
This is grammatically incorrect. Maybe make it similar to the tournaments message ("One of your alts is already in the game.")?